auth/tests/unit/cognito-waf-log-group-kms-key.unit.test.ts

```diff
@@ -0,0 +1,74 @@
+import { Template } from "aws-cdk-lib/assertions";
+import { schema } from "yaml-cfn";
+import { describe, it, beforeAll } from "vitest";
+import { readFileSync } from "fs";
+import { load } from "js-yaml";
+
+let template: Template;
+
+describe("Set up the KMS Key for encrypting the Cognito WAF Log Group for GovUK app", () => {
+  beforeAll(() => {
+    const yamltemplate: string = load(readFileSync("template.yaml", "utf-8"), {
+      schema: schema,
+    });
+    template = Template.fromJSON(yamltemplate);
+  });
+  it("has a description of what the key does", () => {
+    template.hasResourceProperties("AWS::KMS::Key", {
+      Description: "KMS key for encrypting the Cognito WAF Log Group",
+    });
+  });
+  it("has key rotation enabled", () => {
+    template.hasResourceProperties("AWS::KMS::Key", {
+      EnableKeyRotation: true,
+    });
+  });
+  it("has a key policy", () => {
```

Comment:
[Non-blocking, suggestion]: could be good to describe what the key policy is doing in the test description